modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -290,7 +293,7 @@ export default class TileLayer<
   }
 
   renderSubLayers(
-    props: TileLayerProps & {
+    props: _TileLayerProps<DataT> & {
```

Comment:
This is technically incorrect as _TileLayerProps has lots of optional stuff.

And _in_ this stage of processing `props` are resolved.

I this we should invent something like this:

```
export type ComponentProps<T extends Compoennt> = Component["props"]
```


and this line would be:
```
renderSubLayers(
    props: ComponentProps<this>
 ) {
     props.tileSize // is of type `number` with optionals removed
}
```
This would of course remove assume `DataT=any`, but if someone needs genericty, he can write:

```
props: ComponentProps<TileLayer<DataT,ExtraPropsT>> 
```
and have proper signature.